chainer/functions/math/matmul.py

```diff
@@ -244,8 +244,8 @@ def forward_gpu(self, x):
     def backward_gpu(self, x, gy):
         a, b = x
         batch_size = a.shape[0]
-        ga = cuda.empty((batch_size,) + _as_mat(a[0]).shape)
-        gb = cuda.empty((batch_size,) + _as_mat(b[0]).shape)
+        ga = cuda.cupy.empty((batch_size,) + _as_mat(a[0]).shape)
+        gb = cuda.cupy.empty((batch_size,) + _as_mat(b[0]).shape)
```

Comment:
Please set `dtype`. I mean `ga = cuda.cupy.empty(..., a.dtype)`.
Default `dtype` of `cupy.empty` is `float`, is not `float32`.